Adding terraform output trimming changes to HEAD

pswish opened the pull request:
This pull request integrates the terraform Logging changes during the Start Scenario action when triggered by the onclick from the React dashboard. 

1. Why is this change being made?

     - Problems were reported from a classroom test where scenarios took an unacceptable time to start. Reports of 7+ minutes to start were observed.

2. What has changed?

    - Modified the start scenario function in tasks.py in /py_flask/utils/

3. What concerns do you have?
    - Without a classroom test, real world results on this change cannot be realized effectively

4. How was this tested?
    - Tested starting each scenario in a test env.

5. Any related PRs or CRs?

- https://github.com/edurange/edurange3/pull/73
- https://github.com/edurange/edurange3/pull/76
- Added ELF inflection change to eliminate dual provider error

--------------
- Here is what the new output looks like:

![image](https://github.com/user-attachments/assets/9c404c8b-13f6-45da-9add-bbaf045671f7)

- To bypass the change and see the old output, see the comment on the last line:

![image](https://github.com/user-attachments/assets/43db207e-8b11-40ed-aaf3-0a2e6c1e521f)

- Example of the error capture:

![image](https://github.com/user-attachments/assets/dbb06259-7aab-4eef-90c6-396d5b1a2265)

- If you want to add more phrases to the output, add your text string to the list named "important_phrases" here:

![image](https://github.com/user-attachments/assets/83bc93d5-8bd0-4739-a2d2-7539c0b0c078)

cookjackc commented:
When testing - Unable to destroy scenarios.

Steps to reproduce with an empty database:
Create a new scenario
Start it
Stop it
Try to destroy it

Results:
```
  File "/home/edurange/edurange3/py_flask/utils/tasks.py", line 438, in destroy_scenario_task
    NotifyCapture("Failed to delete scenario " + name + ": Scenario files could not be found.")

...

sqlalchemy.exc.DataError: (psycopg2.errors.StringDataRightTruncation) value too long for type character varying(60)

[SQL: INSERT INTO notification (detail, date) VALUES (%(detail)s, %(date)s) RETURNING notification.id]
[parameters: {'detail': 'Failed to delete scenario newStartTF: Scenario files could not be found.', 'date': datetime.datetime(2024, 11, 19, 19, 42, 32, 179995, tzinfo=datetime.timezone.utc)}]
```

This does not occur on main, if I had to guess the chdir is now causing celery to lose its place - unclear why the stop still works but the destroy does not.

pswish commented:
The destroy never worked for me but it has been some time since I tried it. God to know that it is working on main, i'll see if i can see what was missed as all the chdir is clunky.

pswish commented:
Refactored the start_scenario function, and modified the behavior to be cleaner, here is the commit: https://github.com/edurange/edurange3/commit/86999148e8959102ff31972c7a09b3ae32577294

pswish commented:
Closing due to a replacement PR #79
